pkg/maven/maven.go

```diff
@@ -0,0 +1,118 @@
+package maven
+
+import (
+	"bytes"
+
+	"github.com/SAP/jenkins-library/pkg/http"
+	"github.com/SAP/jenkins-library/pkg/log"
+	"io"
+	"strings"
+)
+
+type ExecuteOptions struct {
+	PomPath                     string   `json:"pomPath,omitempty"`
+	ProjectSettingsFile         string   `json:"projectSettingsFile,omitempty"`
+	GlobalSettingsFile          string   `json:"globalSettingsFile,omitempty"`
+	M2Path                      string   `json:"m2Path,omitempty"`
+	Goals                       []string `json:"goals,omitempty"`
+	Defines                     []string `json:"defines,omitempty"`
+	Flags                       []string `json:"flags,omitempty"`
+	LogSuccessfulMavenTransfers bool     `json:"logSuccessfulMavenTransfers,omitempty"`
+	ReturnStdout                bool     `json:"returnStdout,omitempty"`
+}
+
+type mavenExecRunner interface {
+	Stdout(out io.Writer)
+	Stderr(err io.Writer)
+	RunExecutable(e string, p ...string) error
+}
+
+const mavenExecutable = "mvn"
+
+func Execute(options *ExecuteOptions, command mavenExecRunner) (string, error) {
+	stdOutBuf, stdOut := evaluateStdOut(options)
+	command.Stdout(stdOut)
+	command.Stderr(log.Entry().Writer())
+
+	parameters := getParametersFromOptions(options, &http.Client{})
+
+	err := command.RunExecutable(mavenExecutable, parameters...)
+	if err != nil {
+		log.Entry().
+			WithError(err).
+			WithField("command", append([]string{mavenExecutable}, parameters...)).
+			Fatal("failed to execute run command")
+	}
+
+	if stdOutBuf == nil {
+		return "", nil
```

Comment:
Minor: Not sure if I got it correctly, but we could always return standard out and showing the result in the log because of the MultiWriter. I think the reason why we introduced the parameter ReturnStdout was that Jenkins does just support either returning our printing.

Reply:
I agree. The returnStdOut should only be used for piper internal stuff, e.g. ´mvn evaluate´.
I will remove that parameter from the metadata.